waspc/data/Generator/templates/react-app/src/auth/pages/OAuthCallback.tsx

```diff
@@ -0,0 +1,82 @@
+{{={= =}=}}
+import { useEffect, useRef, useState } from "react";
+import { Redirect } from 'react-router-dom'
+import { useAuth } from 'wasp/client/auth'
+import { api } from 'wasp/client/api'
+import { initSession } from 'wasp/auth/helpers/user'
+
+export function OAuthCallbackPage() {
+  const { isLoading, error, user } = useOAuthCallbackHandler();
+  
+  if (user) {
+    return <Redirect to="{= onAuthSucceededRedirectTo =}" />;
+  }
+
+  return (
+    <div style={{
+      display: "flex",
+      alignItems: "center",
+      justifyContent: "center",
+      padding: "4rem",
+    }}>
+      {error && <div style={{
+        color: "rgb(239 68 68)"
+      }}>{error}</div>}
```

Comment:
Inline CSS is the most vanilla way of styling React components (next to using CSS modules, but I wanted to keep everything in one file), so that's perfectly fine.

The `div`s are also okay IMHO since there isn't really any other semantic elements that fit for this use case. 

I've moved the CSS into separate objects to make it a bit more readable :)